src/engine/move_handling.h

```diff
@@ -60,6 +60,7 @@ constexpr void performCastleMove(BitBoard& board, const movegen::Move& move, uin
     }
 }
 
+template<Player player>
 constexpr void performPromotionMove(BitBoard& board, const movegen::Move& move, uint64_t& hash)
 {
     const bool isWhite = board.player == PlayerWhite;
```

Comment:
These can be evaluated at compile time as well now :fire: 
```suggestion
    constexpr bool isWhite = player == PlayerWhite;
    constexpr auto type = isWhite ? WhitePawn : BlackPawn;
    uint64_t& pawns = board.pieces[type];
```